### PR TITLE
Dedicated job for memray tests

### PR DESCRIPTION
**Pull request by fjetter**
Closes https://github.com/dask/distributed/issues/8103

I _think_ this is available on 3.10

"Proper" fix would be https://github.com/conda/conda/issues/8089

### CODE REVIEW

File: continuous_integration/environment-3.10.yaml [hunk 1]
```diff
@@ -19,6 +19,7 @@ dependencies:
   - ipywidgets <8.0.5  # https://github.com/dask/distributed/issues/7688
   - jinja2 >=2.10.3
   - locket >=1.0
+  - memray # Only tested here
```
Comment (hendrikmakait) [inline continuous_integration/environment-3.10.yaml:22]:
```suggestion
  - memray # Only tested here. This does NOT exist on Windows
```

**Comment by hendrikmakait** [comment]
Those are the options I came up with as well. Another less-than-ideal one would be moving it into a dedicated package/repo that could have a lower-complexity CI.

**Comment by fjetter** [comment]
> Those are the options I came up with as well. Another less-than-ideal one would be moving it into a dedicated package/repo that could have a lower-complexity CI.

Don't like this either. I like that this kind of tooling is just packaged and shipped with the core. I think I'll go for the extra CI step ¯\_(ツ)_/¯

File: distributed/diagnostics/tests/test_memray.py [hunk 2]
```diff
@@ -4,6 +4,7 @@
 
 memray = pytest.importorskip("memray")
 
+pytestmark = pytest.mark.extra_packages
```
Comment (fjetter) [inline distributed/diagnostics/tests/test_memray.py:7]: I added this marker to have the job run really quickly. We could also mark with memray explicitly. No strong preference.

**Comment by fjetter** [comment]
FYI I accidentally published the branch to dask/distributed instead of my fork. I suspect this PR will be merged very quickly so I won't migrate the branch. mea culpa

**Pull request merged**